CONVENTIONS.md

```diff
@@ -15,6 +16,12 @@
   - [Identifiers](#identifiers)
   - [Variables](#variables)
 
+## Plugin Location
+
+- Most plugins should start in a repository outside of the OpenSearch project. This allows for rapid iteration during initial development without the overhead of being an official part of the project.
```

Comment:
Nit: Remove most. It's good to be directive. 
Nit: "OpenSearch project" -> opensearch-project ...
Nit: "start" is probably not right, "create" is more appropriate.

I would say "Create new plugins in your personal GitHub account. ..."